comm.c

```diff
@@ -9,41 +9,87 @@
 
 static int comm[2][2] = {{-1, -1}, {-1, -1}};
 
-ssize_t read_comm_request(char **buf_ptr) {
+ssize_t write_string(int fd, const char * const *string, size_t len) {
```

Comment:
I don't have the time and setup to comment on the UI right now, but can advise on matters of C.

A double pointer is unnecessary when writing strings; this function and the functions which call it should use a plain `const char *string` instead.

(Using unnecessary pointers is also risky: the current `write(fd, string[offs], len - offs);` compiles fine, but actually has a bug because `string[offs]` indexes the outer pointer and not the inner one; with a `const char *string` argument, one has `write(fd, &string[offs], len - offs);` instead.)

Reply:
Oh, you are right. I guess my thinking was that "`char *` is a string, and a string should be passed by reference instead of value". But of course it's already a reference because it's a pointer. I fixed this now in bc18cc2 (I can clean up the commit history at the end.)

I also see that `string[offs]` is wrong, but how come it works? ... Oh is it because `offs` always happens to be zero?